src/date/index.ts

```diff
@@ -27,3 +31,378 @@ export function getMonthAbbrByIndex(monthNumber: number): string {
 
   return monthNames[ monthNumber - 1 ];
 }
+
+
+/**
+ * This function converts date input coming from backend (generally in long format (ex. '2022-02-01T12:16:13'))
+ * to a specified format provided in 2nd paramter
+ * It converts all types of dates accepted by dayjs
+ *
+ * @param {Date} longDateValue - Date input to be converted to another format. This generally comes from backend and is
+ * in long format. For example '2022-02-01T12:16:13'
+ * @param {string} dateFormat - Format in which the date needs to be converted to.
+ *
+ * @example
+ * ```
+ * getDateInRequiredFormat('2022-02-01T12:16:13', 'DD MMM, hh:mm A');  // Output will be '01 Feb, 12:16 PM'
+ * getDateInRequiredFormat('2022-01-28T12:54:40', 'DD MMM, hh:mm A');  // Output will be '28 Jan, 12:54 PM'
+ * getDateInRequiredFormat('2022-01-25T12:08:12', 'DD MMM YYYY,  hh:mm A);  // Output will be '25 Jan 2022,  12:08 PM'
+ * getDateInRequiredFormat('2021/11/20', 'YYYY-MM-DD');  // Output will be '2021-11-20'
+ * getDateInRequiredFormat('21 Nov 2020', 'YYYY-MM-DD');  // Output will be '2020-11-21'
+ * getDateInRequiredFormat('Fri Feb 04 2022 15:24:28 GMT+0530 (India Standard Time)', 'YYYY-MM-DD');  // Output will be '2022-02-04'
+ * ```
+ */
+export function getDateInRequiredFormat(longDateValue: Date, dateFormat: string = 'DD MMM YYYY') {
+  try {
+    if (!isEmpty(dateFormat)) {
+      const str = dayjs(longDateValue).format(dateFormat);
+
+      if (str === 'Invalid Date') {
+        return '';
+
+      } else {
+        return str;
+      }
+    }
+
+  } catch (error) {
+    console.error('Error in getDateInRequiredFormat: ', error);
+  }
+}
+
+
+/**
+ * This function returns age as a number from the date of birth as an input
+ *
+ * @param {Date} birthDate - birth date of a person in Date format
+ *
+ * @example
+ * ```
+ * getAgeFromDateOfBirth(new Date());  // Output is ''
+ * getAgeFromDateOfBirth('Sun Oct 10 1993 05:30:00 GMT+0530 (India Standard Time)');  // Output is 28
+ * getAgeFromDateOfBirth('1993-10-10');  // Output is 28
+ * getAgeFromDateOfBirth();  // Output is ''
+ * ```
+ */
+export function getAgeFromDateOfBirth(birthDate: Date) {
+  try {
+    const today = new Date();
+
+    birthDate = new Date(birthDate);
+
+    let age = today.getFullYear() - birthDate.getFullYear();
+    const monthDifference = today.getMonth() - birthDate.getMonth();
+
+    if (monthDifference < 0 || (monthDifference === 0 && today.getDate() < birthDate.getDate())) {
```

Comment:
What are we doing in this condition. Not clear in first look. Can extract in a constant with proper name

Reply:
updated